src/OpenTelemetry.Exporter.InMemory/InMemoryExporterMetricsExtensions.cs

```diff
@@ -93,5 +135,38 @@ public static MeterProviderBuilder AddInMemoryExporter(this MeterProviderBuilder
 
             return builder.AddReader(metricReader);
         }
+
+        private static MeterProviderBuilder AddInMemoryExporter(
+            MeterProviderBuilder builder,
+            ICollection<MetricSnapshot> exportedItems,
+            MetricReaderOptions metricReaderOptions,
+            Action<MetricReaderOptions> configureMetricReader)
+        {
+            configureMetricReader?.Invoke(metricReaderOptions);
+
+            var metricExporter = new InMemoryExporter<Metric>(
+                exportFunc: metricBatch =>
```

Comment:
I don't think the `Batch<Metric>` is passed here with `in` modifier constraint.

Reply:
func doesn't support the `in` keyword, but the encapsulated method still uses `in`:
https://github.com/open-telemetry/opentelemetry-dotnet/blob/04b894e1fd1d6580ff4adf575bc1eb7fe5254446/src/OpenTelemetry.Exporter.InMemory/InMemoryExporter.cs#L39

I just debugged and confirmed that any changes made to the batch within the func don't persist. 
I think I can author a test that validates that the access modifier still applies here.

Would that address your concern here?

Reply:
The main benefit of using `in` here is to avoid copying the `Batch` struct. If it's not passed with the `in` modifier, the method would get a copy of the struct.